cpp/include/cudf/io/detail/tokenize_json.hpp

```diff
@@ -134,7 +134,8 @@ std::pair<rmm::device_uvector<PdaTokenT>, rmm::device_uvector<SymbolOffsetT>> ge
   device_span<SymbolT const> json_in,
   cudf::io::json_reader_options const& options,
   rmm::cuda_stream_view stream,
-  rmm::device_async_resource_ref mr);
+  rmm::device_async_resource_ref mr,
+  bool remove_line_end_token = true);
```

Comment:
By cudf convention, `stream` and `mr` are the two last parameters. So `remove_line_end_token` should come before them. But by having so, we cannot set the default parameter `true` for it.

Reply:
It applies to other calls as well where `mr` is not present.

Reply:
Got it, I've added `LineEndTokenOption` enum to keep/discard LineEnd tokens. 
Additionally, I've introduced another declaration for get_token_stream to maintain backward compatibility.